scripts/init.sh

```diff
@@ -0,0 +1,42 @@
+nvm_shell_config='
+  export NVM_DIR="$HOME/.nvm"\n
+  [ -s "/opt/homebrew/opt/nvm/nvm.sh" ] && \. "/opt/homebrew/opt/nvm/nvm.sh"  # This loads nvm\n
+  [ -s "/opt/homebrew/opt/nvm/etc/bash_completion.d/nvm" ] && \. "/opt/homebrew/opt/nvm/etc/bash_completion.d/nvm"  # This loads nvm bash_completion\n
+'
+
+[ -s "/opt/homebrew/opt/nvm/nvm.sh" ] && \. "/opt/homebrew/opt/nvm/nvm.sh" # This loads nvm
+
+ensure_nvm() {
+  if ! command -v nvm &>/dev/null; then
+    brew install nvm
+    if [ -n "$ZSH_VERSION" ]; then
+      echo $nvm_shell_config >>~/.zshrc
+    elif [ -n "$BASH_VERSION" ]; then
+      echo $nvm_shell_config >>~/.bashrc
+    fi
+  fi
+
+  nvm install
+}
+
+ensure_ruby() {
```

Comment:
this will only work on macos b/c of homebrew. readme should probably say something about this or we should also consider into a linux solution

Reply:
i guess pods don't work on linux, but running web/desktop should be good

Reply:
yeah given all the complexity here im thinking just a script to check versions for now may be the move?